### PR TITLE
#161656449 Fix room selection issue

### PR DESCRIPTION
**Pull request by Stylll**
#### What does this PR do?
- Fixes the issue with room selection based on gender issue

#### Description of Task to be completed?
- ensure that requesters cannot book for rooms occupied by the opposite gender
- ensure that beds not existing in the selected destination guesthouse cannot be booked

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Checkout the branch `git checkout bg-fix-bed-selection-161656449`
- Install dependencies using `yarn install`
- Rollback and run database migrations with `yarn db:rollmigrate`
- Start the server with `make start`
- Setup the frontend app with this - `https://github.com/andela/travel_tool_front`
- Visit https://travela-local.andela.com
- Login to the application
- Create a request setting your gender as `Male`
- Take note of the bed and room you selected
- Create another request with the same date range as the previous but setting your gender as `Female`
- The list of available rooms should not consist of any bed in the room which you selected in the previous request.

#### Any background context you want to provide?
- You should have available rooms in a guesthouse in order to test this bug fix

#### What are the relevant pivotal tracker stories?
[#161656449](https://www.pivotaltracker.com/story/show/161656449)

### CODE REVIEW

File: src/modules/requests/__tests__/requests.test.js [hunk 1]
```diff
@@ -819,6 +863,21 @@ describe('Requests Controller', () => {
           });
       });
 
+      it('should return error if bed id does not exist in any guesthouse', async (done) => {
```
Comment (loicemeyo) [inline src/modules/requests/__tests__/requests.test.js]: @Stylll the `it` block for both tests is the same but I see that you're testing for different instances. Please update this

Comment (Stylll) [reply src/modules/requests/__tests__/requests.test.js]: Hi @loicemeyo I've adjusted the it blocks. Kindly review.

File: src/modules/requests/RequestsController.js [hunk 2]
```diff
@@ -205,6 +205,7 @@ class RequestsController {
       params.limit,
       getTotalCount(params.status, count),
     );
+    console.log('requests:', requests.rows.trips);
```
Comment (amarachukwu-agbo) [inline src/modules/requests/RequestsController.js]: Please remove console statement

Comment (Stylll) [reply src/modules/requests/RequestsController.js]: @amarachukwu-agbo Thanks for spotting this. It has been removed.

**Comment by amarachukwu-agbo** [comment]
@Stylll , good job, works as expected. @samuelkubai , LGTM

**Pull request merged**